File: Hacker Rank Solutions Better Explained/Triple_Sum.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int p,q,r,i,x,y;
    cin>>p>>q>>r;

    int a[p],b[q],c[r];
    //for condition
    for(i=0;i<p;++i) {
        //>>
        cin>>a[i];
    }
    //for condition
    for(i=0;i<q;++i) {
        //>>
        cin>>b[i];
    }
    //for condition
    for(i=0;i<r;++i) {
        //>>
        cin>>c[i];
    }
    //apply function
    sort(a,a+p);
    //apply function
    sort(b,b+q);
    //apply function
    sort(c,c+r);
    //long
    long distinct_a[p],distinct_b[q],distinct_c[r];
    //set<int>
    set<int> s;
    //for condition
    for(i=0;i<p;++i) {
        //apply function
        s.insert(a[i]);
        //set to
        distinct_a[i]=s.size();
    }
    //apply function
    s.clear();
    //for condition
    for(i=0;i<q;++i) {
        //apply function
        s.insert(b[i]);
        //set to
        distinct_b[i]=s.size();
    }
    //apply function
    s.clear();
    //for condition
    for(i=0;i<r;++i) {
        //apply function
        s.insert(c[i]);
        //set to
        distinct_c[i]=s.size();
    }
    //set to
    long ans=0;
    //set to apply function
    x = upper_bound(a,a+p,b[0])-a;
    //set to apply function
    y = upper_bound(c,c+r,b[0])-c;
    //subtract by
    x-=1,y-=1;
    //if condition
    if(x>=0 && y>=0) {
        //augment by
        ans += distinct_a[x]*distinct_c[y];
    }
    //for condition
    for(i=1;i<q;++i) {
        //if condition
        if(b[i]!=b[i-1]) {
            //set to apply function
            x = upper_bound(a,a+p,b[i])-a;
            //set to apply function
            y = upper_bound(c,c+r,b[i])-c;
            //subtract by
            x-=1,y-=1;
            //if condition
            if(x>=0 && y>=0) {
                //augment by
                ans += distinct_a[x]*distinct_c[y];
            }
        }
    }
    //<<
    cout<<ans<<endl;
    return 0;
}

File: Hacker Rank Solutions Better Explained/TripleSum.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Solution {

    // int[], int[], int[]
    static long triplets(int[] a, int[] b, int[] c)
    {
        //int[] = int[].Distinct().ToArray(), set to apply functions
        a = a.Distinct().ToArray();
        b = b.Distinct().ToArray();
        c = c.Distinct().ToArray();
        //Array.Sort() Int[] apply functions        
        Array.Sort(a);
        Array.Sort(b);
        Array.Sort(c);
        //Int set to
        var result = 0L;
        //foreach(int int[]) loop through
        foreach (var item in b)
        {
            //int = Count(int[],int) set to function
            var countA = Count(a, item);
            //int = Count(int[],int) set to function
            var countB = Count(c, item);
            //int = int * int augment by
            result += countA * countB;
        }
        //int
        return result;
    }
    //int[], int
    static int Count(int[] array, int item)
    {
        //int = Array.BinarySearch(int[], int)) set to calculate
        var searchIndex = Array.BinarySearch(array, item);
        //if(int >= 0) return int + 1 if condition return int
        if( searchIndex >= 0) return searchIndex + 1;
        //else return Math.Abs(int)-1 
        else return Math.Abs(searchIndex) - 1;
    }
    
    static void Main(string[] args) {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        string[] lenaLenbLenc = Console.ReadLine().Split(' ');

        int lena = Convert.ToInt32(lenaLenbLenc[0]);

        int lenb = Convert.ToInt32(lenaLenbLenc[1]);

        int lenc = Convert.ToInt32(lenaLenbLenc[2]);

        int[] arra = Array.ConvertAll(Console.ReadLine().Split(' '), arraTemp => Convert.ToInt32(arraTemp))
        ;

        int[] arrb = Array.ConvertAll(Console.ReadLine().Split(' '), arrbTemp => Convert.ToInt32(arrbTemp))
        ;

        int[] arrc = Array.ConvertAll(Console.ReadLine().Split(' '), arrcTemp => Convert.ToInt32(arrcTemp))
        ;
        long ans = triplets(arra, arrb, arrc);

        textWriter.WriteLine(ans);

        textWriter.Flush();
        textWriter.Close();
    }
}
